sympy/physics/mechanics/tests/test_kane3.py

```diff
@@ -257,35 +284,54 @@ def test_bicycle():
     # many rows as *total* coordinates and speeds, but only as many columns as
     # independent coordinates and speeds.
 
-    forcing_lin = KM.linearize()[0]
-
+    A, B, _ = KM.linearize(
+        A_and_B=True,
+        op_point={
+            # Operating points for the accelerations are required for the
+            # linearizer to eliminate u' terms showing up in the coefficient
+            # matrices.
+            u1.diff(): 0,
+            u2.diff(): 0,
+            u3.diff(): 0,
+            u4.diff(): 0,
+            u5.diff(): 0,
+            u6.diff(): 0,
+            u1: 0,
+            u2: 0,
+            u3: v / PaperRadRear,
+            u4: 0,
+            u5: 0,
+            u6: v / PaperRadFront,
+            q1: 0,
+            q2: 0,
+            q4: 0,
+            q5: 0,
+        },
+        linear_solver=cramer_solve,
+    )
     # As mentioned above, the size of the linearized forcing terms is expanded
     # to include both q's and u's, so the mass matrix must have this done as
     # well.  This will likely be changed to be part of the linearized process,
     # for future reference.
-    MM_full = KM.mass_matrix_full
-
-    MM_full_s = msubs(MM_full, val_dict)
-    forcing_lin_s = msubs(forcing_lin, KM.kindiffdict(), val_dict)
+    A_s = A.xreplace(val_dict)
+    B_s = B.xreplace(val_dict)
 
-    MM_full_s = MM_full_s.evalf()
-    forcing_lin_s = forcing_lin_s.evalf()
+    A_s = A_s.evalf()
+    B_s = B_s.evalf()
 
     # Finally, we construct an "A" matrix for the form xdot = A x (x being the
     # state vector, although in this case, the sizes are a little off). The
     # following line extracts only the minimum entries required for eigenvalue
     # analysis, which correspond to rows and columns for lean, steer, lean
     # rate, and steer rate.
-    Amat = MM_full_s.inv() * forcing_lin_s
-    A = Amat.extract([1, 2, 4, 6], [1, 2, 3, 5])
+    A = A_s.extract([1, 2, 3, 5], [1, 2, 3, 5])
```

Comment:
Why is it now `5` and not `6` like previously?

Reply:
I can't remember exactly, but I believe the new linearizer gave a different result than the old one and it necessitated this change. I'll have to investigate again. Maybe I noted something in one of the commit messages.

Reply:
Here are the states stored in KanesMethod and then the size of the A matrix:

```
ipdb> KM.q
Matrix([
[q1(t)],
[q2(t)],
[q5(t)],
[q4(t)]])
ipdb> KM.u
Matrix([
[u2(t)],
[u3(t)],
[u5(t)],
[u1(t)],
[u4(t)],
[u6(t)]])
ipdb> A_s.shape
(6, 6)
```

and the declaration of KanesMethod:

```
    KM = KanesMethod(N, q_ind=[q1, q2, q5],
            q_dependent=[q4], configuration_constraints=conlist_coord,
            u_ind=[u2, u3, u5],
```

So I guess the rows/cols of A_s correspond to q1, q2, q5, u2, u3, u5. 

and the speeds are:

```
    # Speeds are: yaw frame ang. rate, roll frame ang. rate, rear wheel frame
    # ang.  rate (spinning motion), frame ang. rate (pitching motion), steering
    # frame ang. rate, and front wheel ang. rate (spinning motion).
    u1, u2, u3, u4, u5, u6 = dynamicsymbols('u1 u2 u3 u4 u5 u6')
```

so

u1 = yaw rate
u2 = roll rate
u3 = rear wheel rate
u4 = pitch rate
u5 = steer rate
u6 = front wheel rate

that means that we need indices: 1, 2, 3 and 5, which is what we have.

Reply:
No idea why it was 6 before. I'd have to roll back to a really old sympy to figure it out.